src/Compilers/CSharp/Test/Semantic/Semantics/AmbiguousOverrideTests.cs

```diff
@@ -136,13 +137,25 @@ public class Derived2 : Derived<int>
             var comp = CreateCompilation(text3, ref2, assemblyName: "Test3");
             var diagnostics = comp.GetDiagnostics();
 
-            comp.VerifyDiagnostics(
-                // (4,26): error CS0462: The inherited members 'Derived<TInt>.Method(long, TInt)' and 'Derived<TInt>.Method(long, int)' have the same signature in type 'Derived2', so they cannot be overridden
-                //     public override void Method(long l, int i) { }  //CS0462 and CS1957
-                Diagnostic(ErrorCode.ERR_AmbigOverride, "Method").WithArguments("Derived<TInt>.Method(long, TInt)", "Derived<TInt>.Method(long, int)", "Derived2").WithLocation(4, 26),
-                // (4,26): warning CS1957: Member 'Derived2.Method(long, int)' overrides 'Derived<int>.Method(long, int)'. There are multiple override candidates at run-time. It is implementation dependent which method will be called.
-                //     public override void Method(long l, TInt i) { }
-                Diagnostic(ErrorCode.WRN_MultipleRuntimeOverrideMatches, "Method").WithArguments("Derived<int>.Method(long, int)", "Derived2.Method(long, int)").WithLocation(4, 26));
+            if (comp.Assembly.RuntimeSupportsDefaultInterfaceImplementation)
+            {
+                comp.VerifyDiagnostics(
+                    // (4,26): error CS0462: The inherited members 'Derived<TInt>.Method(long, TInt)' and 'Derived<TInt>.Method(long, int)' have the same signature in type 'Derived2', so they cannot be overridden
+                    //     public override void Method(long l, int i) { }  //CS0462 and CS1957
+                    Diagnostic(ErrorCode.ERR_AmbigOverride, "Method").WithArguments("Derived<TInt>.Method(long, TInt)", "Derived<TInt>.Method(long, int)", "Derived2").WithLocation(4, 26)
+                    );
+            }
+            else
+            {
+                comp.VerifyDiagnostics(
+                    // (4,26): error CS0462: The inherited members 'Derived<TInt>.Method(long, TInt)' and 'Derived<TInt>.Method(long, int)' have the same signature in type 'Derived2', so they cannot be overridden
+                    //     public override void Method(long l, int i) { }  //CS0462 and CS1957
+                    Diagnostic(ErrorCode.ERR_AmbigOverride, "Method").WithArguments("Derived<TInt>.Method(long, TInt)", "Derived<TInt>.Method(long, int)", "Derived2").WithLocation(4, 26),
+                    // (4,26): warning CS1957: Member 'Derived2.Method(long, int)' overrides 'Derived<int>.Method(long, int)'. There are multiple override candidates at run-time. It is implementation dependent which method will be called. Please use a newer runtime.
+                    //     public override void Method(long l, TInt i) { }
```

Comment:
>//     public override void Method(long l, TInt i) { } [](start = 20, length = 54)

It looks like a warning from a different compilation made its way into diagnostics. Any idea how did this happen?

Reply:
This is correct. It is precisely the same diagnostic from before this PR.

---
In reply to: [446335378](https://github.com/dotnet/roslyn/pull/44025#discussion_r446335378) [](ancestors = 446335378)

Reply:
> This is correct. It is precisely the same diagnostic from before this PR.

Any idea how did this happen? 

---
In reply to: [446408459](https://github.com/dotnet/roslyn/pull/44025#discussion_r446408459) [](ancestors = 446408459,446335378)

Reply:
I'm not sure what you mean. You're wondering why we report the same diagnostics that we always did?

---
In reply to: [446411091](https://github.com/dotnet/roslyn/pull/44025#discussion_r446411091) [](ancestors = 446411091,446408459,446335378)

Reply:
> I'm not sure what you mean. You're wondering why we report the same diagnostics that we always did?

Yes, I am wondering if you understand why was this happening. This is not a trick question.


---
In reply to: [446412205](https://github.com/dotnet/roslyn/pull/44025#discussion_r446412205) [](ancestors = 446412205,446411091,446408459,446335378)

Reply:
Yes, because we don't (and never did) suppress the warning in the face of this particular error.

---
In reply to: [446414161](https://github.com/dotnet/roslyn/pull/44025#discussion_r446414161) [](ancestors = 446414161,446412205,446411091,446408459,446335378)

Reply:
> Yes, because we don't (and never did) suppress the warning in the face of this particular error.

Let me rephrase the question. I am wondering if you understand why a warning about code in a different compilation and pointing at location in a different compilation is reported for this compilation.

---
In reply to: [446414685](https://github.com/dotnet/roslyn/pull/44025#discussion_r446414685) [](ancestors = 446414685,446414161,446412205,446411091,446408459,446335378)

Reply:
Because that is the way the diagnostic is produced in the compiler. I presume the intent was to produce the diagnostic where the user was most likely to be able to fix it (by renaming, for example). None of that has anything to do with any compiler changes in the present PR.

---
In reply to: [446419715](https://github.com/dotnet/roslyn/pull/44025#discussion_r446419715) [](ancestors = 446419715,446414685,446414161,446412205,446411091,446408459,446335378)